ambry-network/src/main/java/com/github/ambry/network/NettyServerRequestResponseChannel.java

```diff
@@ -56,11 +57,18 @@ public NettyServerRequestResponseChannel(NetworkConfig config, Http2ServerMetric
         networkRequestQueue::numDroppedRequests);
   }
 
-  /** Send a request to be handled */
+  /**
+   * Send a request to be handled
+   * @return {@code True} if we are able to send the request over the channel. Else {@code False}
+   */
   @Override
-  public void sendRequest(NetworkRequest request) throws InterruptedException {
-    networkRequestQueue.offer(request);
-    http2ServerMetrics.requestEnqueueTime.update(System.currentTimeMillis() - request.getStartTimeInMs());
+  public boolean sendRequest(NetworkRequest networkRequest) throws InterruptedException {
```

Comment:
Does this fn take requests coming in from the frontend and add it to a queue in the server ?
What are the threads in the server consuming this queue ?